src/index.tsx

```diff
@@ -81,8 +81,49 @@ const FastUrlModule: FastUrlNative = {
 export class URLSearchParams {
   private _urlSearchParams: ReturnType<FastUrlNative['URLSearchParams']>;
 
-  constructor(url?: string) {
-    this._urlSearchParams = FastUrlModule.URLSearchParams(url ?? '');
+  /**
+   * URLSearchParams accepts a variety of different input types:
+   *
+   * - A string, which will be parsed from application/x-www-form-urlencoded format. A leading '?' character is ignored.
+   * - A literal sequence of name-value string pairs, or any object — such as a FormData object — with an iterator that
+   *   produces a sequence of string pairs. Note that File entries will be serialized as [object File] rather than as
+   *   their filename (as they would in an application/x-www-form-urlencoded form).
+   * - A record of string keys and string values. Note that nesting is not supported.
+   *
+   * See https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams
+   */
+  constructor(
+    init:
+      | string
+      | Record<string, string>
+      | IterableIterator<[string, string]> = ''
+  ) {
```

Comment:
Now that I'm thinking about it it's probably better to avoid the overhead of appending through JSI for every entry here - it should probably be written on the C++ side of things

Reply:
idk if I have the C++ chops to implement the iterator part in c++ though

Reply:
To implement iterator part in C++ side will be tricky cuz I don't see any iterator things in jsi.h for now, and I think the performance here was not that bad.